### PR TITLE
Return all gdb output

### PR DESCRIPTION
**Pull request by Avgor46**

### CODE REVIEW

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -276,7 +276,7 @@ impl<'a> GdbCommand<'a> {
         self.ex("p/x $_siginfo")
     }
 
-    /// Execute gdb and get result for each command.
+    /// Execute gdb and get result without raw stderr
```
Comment (anfedotoff) [inline src/lib.rs:279]: from raw stdout

File: src/lib.rs [hunk 2]
```diff
@@ -286,6 +286,33 @@ impl<'a> GdbCommand<'a> {
...
+        // Split stdout into lines.
+        let output = String::from_utf8_lossy(&stdout);
+
+        let mut results = self.gdb_result(&output)?;
+        results.push(output.to_string());
+
+        Ok(results)
+    }
+
+    /// Get result for each command.
```
Comment (anfedotoff) [inline src/lib.rs:311]: Result for each executed gdb command from raw gdb output

**Review by anfedotoff — CHANGES_REQUESTED** [review]

File: src/lib.rs [hunk 3]
```diff
@@ -286,6 +286,33 @@ impl<'a> GdbCommand<'a> {
...
+
+        self.gdb_result(&output)
+    }
+
+    /// Execute gdb and get result with raw stderr
+    /// # Return value.
+    ///
+    /// The return value is a vector of strings for each command executed
+    /// and a stderr at the last element of the vector
+    pub fn launch_with_stderr(&self) -> error::Result<Vec<String>> {
```
[the remark this answers is not in the document]
Comment (anfedotoff) [reply src/lib.rs:298]: This function is not  needed. We could use  `launch` or raw+gdb_result(parse) if we need raw stdout.

**Review by anfedotoff — APPROVED** [review]

File: src/lib.rs [hunk 4]
```diff
@@ -286,7 +286,16 @@ impl<'a> GdbCommand<'a> {
 
         // Split stdout into lines.
         let output = String::from_utf8_lossy(&stdout);
-        let lines: Vec<String> = output.lines().map(|l| l.to_string()).collect();
+
+        self.parse(output)
+    }
+
+    /// Result for each executed gdb command from raw gdb output.
```
Comment (SweetVishnya) [inline src/lib.rs:293]: Parse result

File: src/lib.rs [hunk 5]
```diff
@@ -276,7 +276,7 @@ impl<'a> GdbCommand<'a> {
         self.ex("p/x $_siginfo")
     }
 
-    /// Execute gdb and get result for each command.
+    /// Execute gdb and get result from raw stdout
```
[the remark this answers is not in the document]
Comment (Avgor46) [reply src/lib.rs]: done

**Review by anfedotoff — APPROVED** [review]

**Review by SweetVishnya — APPROVED** [review]

**Pull request merged**